File: webhook.php
<?php

$token = "token_bot";
$url_webhook = "https://yourdomain.com/bot.php";

echo "fix endpoint Rubika\n";
$endpoints = [
    "ReceiveUpdate",
    "ReceiveInlineMessage",
    "ReceiveQuery",
    "GetSelectionItem",
    "SearchSelectionItems"
];
foreach ($endpoints as $endpoint) {
    $data = [
        "url" => $url_webhook,
        "type" => $endpoint
    ];

    try {
        $url = "https://botapi.rubika.ir/v3/" . $token . "/" . "updateBotEndpoints";
        $ch = curl_init();
        curl_setopt($ch, CURLOPT_URL, $url);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
        curl_setopt($ch, CURLOPT_HTTPHEADER, ['Content-Type: application/json']);
        curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($data));
        $raw = curl_exec($ch);
        $response = json_decode($raw);

        echo $endpoint . ":\n";

        if (isset($response->status) && $response->status === "OK") {
            $statusText = isset($response->data->status) ? $response->data->status : "unknown";
            echo "   ✅ done - status: " . $statusText . "\n";
        } else {
            echo "   ❌ error - response: " . json_encode($response) . "\n";
        }

    } catch (\Exception $e) {
        echo $endpoint . ":\n";
        echo "   ❌ error Network: " . $e->getMessage() . PHP_EOL . "\n";
    }

    usleep(500000);
}

echo "the end!";

?>

File: webhook.py
import requests
import time

token = "token_bot"
webhook_url = "https://yourdomain.com/bot.php"
api_url = f"https://botapi.rubika.ir/v3/{token}/updateBotEndpoints"


endpoints = [
    "ReceiveUpdate",
    "ReceiveInlineMessage", 
    "ReceiveQuery",
    "GetSelectionItem",
    "SearchSelectionItems"
]

headers = {
    "Content-Type": "application/json"
}

print("setting up Rubika endpoints ...")

for endpoint in endpoints:
    data = {
        "url": webhook_url,
        "type": endpoint
    }
    
    try:
        response = requests.post(api_url, json=data, headers=headers, timeout=30).json()
        
        print(f"{endpoint}:")
        if response.get('status') == 'OK':
            print(f"   ✅ done - status: {response.get('data', {}).get('status', 'Unknown')}")
        else:
            print(f"   ❌ error - response: {response}")
            
    except Exception as e:
        print(f"{endpoint}:")
        print(f"   ❌ error Network: {e}")
    
    time.sleep(0.5)  

print("the end!")

